submissions/api.py

```diff
@@ -89,6 +90,15 @@ def __init__(self, msg="", field_errors=None):
         )
 
 
+# For API stability and low memory usage, large result sets are returned using namedtuples.
+# They take up only as much memory as tuples, but fields are accessible by name,
+# so we can change the fields in the future if needed
+LatestSubmission = namedtuple(
```

Comment:
I've removed the named tuple and used dictionaries instead, as do all other functions in the API.  The only use of the named tuple was in the newly added `get_all_submissions()` function, which is a lazy generator anyway, so it is easy for clients to just extract the data they need while iterating over the result, thereby avoiding excessive memory consumption.